### ISSUE TITLE
v4.0.0 Gluetun

### ISSUE THREAD

**Issue by qdm12** [qdm12/gluetun#1113]
### What's the feature 🧐

There are several pending breaking changes that will simplify the code and allow for faster development.
There is no fixed date really for a v4.0.0 release, but it will ultimately happen.

Changes to be made are listed here:

- Remove `FIREWALL_DEBUG` in favor of `LOG_LEVEL=debug`
- Remove Surfshark region retro compatibility (search for `// TODO .*v4`) and remove `RetroLoc` from server model
- Split `DOT_PRIVATE_ADDRESS` in two: ips and cidr ranges
- Rename `UNBLOCK` to a better name
- ProtonVPN: remove `name` field from updater code
- Throw out v0 http server
- MAYBE have the http server as json rpc instead of json rest?

**Comment by ksurl** [comment]
> Rename `UNBLOCK`

How about `DOMAIN_ALLOW_LIST`

**Comment by moookino** [comment]
can you add hostname resolution for VPN_ENDPOINT instead of IP address?

**Comment by qdm12** [comment]
@moookino this is out of scope, see issue #788 - hiding your comment 😉

**Comment by burnbrigther** [comment]
How about ExpressVPN lightway_udp support?  Please? :)

**Comment by iaacautomation** [comment]
On Kubernetes there is a always a good internal DNS setup, so having a way to completely disable the DNS server would be useful. I think it was removed recently, which takes control away from user.

**Comment by qdm12** [comment]
@iaacautomation please voice your concern on #3279 not here. And answer that question over at #3279 

> Using the Kubernetes DNS server would mean all your DNS traffic would leak out the VPN tunnel. The built-in DNS server now "smartly" forwards requests for local names to whatever nameservers were defined in /etc/resolv.conf at container start, effectively it should forward local name requests to the Kubernetes DNS server, whilst still forwarding public name requests through the VPN tunnel. What's missing from this approach?

Marking both your commment and this comment as off-topic.
